src/dsp_tools/utils/xmlupload/upload_stashed_resptr_props.py

```diff
@@ -97,6 +74,125 @@ def upload_stashed_resptr_props(
     return nonapplied_resptr_props
 
 
+def _upload_all_resptr_props_of_single_resource(
+    resource_in_triplestore: dict[str, Any],
+    link_prop: XMLProperty,
+    res_iri: str,
+    stashed_resource: XMLResource,
+    resptrs: list[str],
+    id2iri_mapping: dict[str, str],
+    con: Connection,
+    nonapplied_resptr_props: dict[XMLResource, dict[XMLProperty, list[str]]],
+    verbose: bool,
+) -> dict[XMLResource, dict[XMLProperty, list[str]]]:
+    """
+    This function takes one resource stashed resource and resptr-props that are specific to a property.
+    It uploads them to the triplestore and removes them from the nonapplied_resptr_props dictionary.
+
+    Args:
+        resource_in_triplestore: The resource retried from the triplestore
+        link_prop: the link property object
+        res_iri: the IRI as given by the DSP-API
+        stashed_resource: The resource from the stash
+        resptrs: List with all the resptr-props for that one property
+        id2iri_mapping: mapping of internal id to the IRI from the DSP-API
+        con: Connection to the DSP-API
+        nonapplied_resptr_props: stashed resources
+        verbose: If True, more information is logged
+
+    Returns:
+        The nonapplied_resptr_props dictionary with the uploaded resource removed
+    """
+    for resptr in resptrs.copy():
+        jsondata = _create_resptr_prop_json_object_to_update(
+            stashed_resource=stashed_resource,
+            resource_in_triplestore=resource_in_triplestore,
+            property_name=resptr,
+            link_prop=link_prop,
+            res_iri=res_iri,
+            id2iri_mapping=id2iri_mapping,
+        )
+        try:
+            try_network_action(con.post, route="/v2/values", jsondata=jsondata)
+        except BaseError as err:
+            # print the message to keep track of the cause for the failure.
+            # Apart from that, no action is necessary:
+            # this resource will remain in nonapplied_resptr_props, which will be handled by the caller
+            orig_err_msg = err.orig_err_msg_from_api or err.message
+            err_msg = f"Unable to upload the resptr prop of '{link_prop.name}' of resource '{stashed_resource.id}'."
+            print(f"    WARNING: {err_msg} Original error message: {orig_err_msg}")
+            logger.warning(err_msg, exc_info=True)
+            continue
+        if verbose:
+            print(f'  Successfully uploaded resptr-prop of "{link_prop.name}". Value: {resptr}')
+            logger.info(f'Successfully uploaded resptr-prop of "{link_prop.name}". Value: {resptr}')
+        nonapplied_resptr_props[stashed_resource][link_prop].remove(resptr)
+    return nonapplied_resptr_props
+
+
+def _unable_to_retrieve_resource(
```

Comment:
As already discussed, I'm not yet convinced of this approach. We have dozens of print and logging statements, every time specific for each case. Should we every time make a new function out of them? Perhaps @BalduinLandolt can give us some advice how this should best be solved. 

If you want to keep it, then I would rename the function. Functions should describe what they do (with a verb), and this one logs/prints if the resource cannot be retrieved, so I suggest `_log_if_unable_to_retrieve_resource()` or something similar.

Reply:
I agree with the name change.

I know it is a bit of a weird approach. The reason, why I extract them in some cases is, because when I want to revise the code, unless I am particularly interested in the logging and so on, I really don't care to see these. And in larger functions where a lot of logging occurs it inflates the function and makes them, in my opinion, very messy and hard to read.

Reply:
I see both sides, here. And I don't have a solution for it either. I might go as far as to rename it to `__warn()` and I definitely would change the call
```python
            _unable_to_retrieve_resource(
                error=err,
                resource=resource,
            )
```
to
```python
            _unable_to_retrieve_resource(err, resource)
```
so it doesn't get in the way as much.

Also, I would remove the docstrings and inline comments from this method. They don't add value, and they add 11 lines of code for a method that consists of 7 lines.

I think, the underlying issue is twofold here: Our error model is not precise enough, so that it is necessary to create informative error messages manually, from the error messages of the exceptions. And secondly, the concepts of logging and user feedback in the console are still not separated properly. Why print errors to the console? One could have an error counter in the background, display a progress bar, in the end print "XYZ finished with 5 errors. See the logs for more information."  
But this is an issue for another day, and should not concern us here in this PR.

Reply:
I made the changes.

However, I am unable to leave out the docstring because of the linters.

Reply:
Then we will have to change the linting settings. It can not be that private methods *require* docstrings. That's not linting, that's tyranny! 😉

Reply:
I looked it up and it ignores private methods by default. I didn't know that, so in this case I was able to delete the docstring.